cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/template/etl/common/PipelineRegisterer.java

```diff
@@ -120,11 +119,18 @@ public Pipeline registerPlugins(ETLConfig config, Class errorDatasetType, Datase
       pluginNum++;
     }
 
-    List<String> sinkPluginIds = new ArrayList<>();
+    List<SinkInfo> sinksInfo = new ArrayList<>();
     List<PipelineConfigurable> sinks = new ArrayList<>();
     for (ETLStage sinkConfig : sinkConfigs) {
       String sinkPluginId = PluginID.from(Constants.Sink.PLUGINTYPE, sinkConfig.getName(), pluginNum).getID();
-      sinkPluginIds.add(sinkPluginId);
+      // create error dataset for sink - if the pipeline is batch and error dataset is configured.
+      // batch uses datasetType TimePartitionedFileSet.
+      if (sinkConfig.getErrorDatasetName() != null &&
```

Comment:
instead of checking the class, can pass a boolean into the constructor that indicates whether sinks can have error datasets or not.